flow/display_list.h

```diff
@@ -0,0 +1,427 @@
+                            SkScalar myt,
+                            SkScalar px,
+                            SkScalar py,
+                            SkScalar pt) = 0;
+
+  virtual void clipRect(const SkRect& rect, bool isAA, SkClipOp clip_op) = 0;
+  virtual void clipRRect(const SkRRect& rrect, bool isAA, SkClipOp clip_op) = 0;
+  virtual void clipPath(const SkPath& path, bool isAA, SkClipOp clip_op) = 0;
+
+  virtual void drawPaint() = 0;
+  virtual void drawColor(SkColor color, SkBlendMode mode) = 0;
+  virtual void drawLine(const SkPoint& p0, const SkPoint& p1) = 0;
+  virtual void drawRect(const SkRect& rect) = 0;
+  virtual void drawOval(const SkRect& bounds) = 0;
+  virtual void drawCircle(const SkPoint& center, SkScalar radius) = 0;
+  virtual void drawRRect(const SkRRect& rrect) = 0;
+  virtual void drawDRRect(const SkRRect& outer, const SkRRect& inner) = 0;
+  virtual void drawPath(const SkPath& path) = 0;
+  virtual void drawArc(const SkRect& bounds,
+                       SkScalar start,
+                       SkScalar sweep,
+                       bool useCenter) = 0;
+  virtual void drawPoints(SkCanvas::PointMode mode,
+                          uint32_t count,
+                          const SkPoint pts[]) = 0;
+  virtual void drawVertices(const sk_sp<SkVertices> vertices,
+                            SkBlendMode mode) = 0;
+  virtual void drawImage(const sk_sp<SkImage> image,
+                         const SkPoint point,
+                         const SkSamplingOptions& sampling) = 0;
+  virtual void drawImageRect(const sk_sp<SkImage> image,
+                             const SkRect& src,
+                             const SkRect& dst,
+                             const SkSamplingOptions& sampling) = 0;
+  virtual void drawImageNine(const sk_sp<SkImage> image,
+                             const SkIRect& center,
+                             const SkRect& dst,
+                             SkFilterMode filter) = 0;
+  virtual void drawImageLattice(const sk_sp<SkImage> image,
+                                const SkCanvas::Lattice& lattice,
+                                const SkRect& dst,
+                                SkFilterMode filter) = 0;
+  virtual void drawAtlas(const sk_sp<SkImage> atlas,
+                         const SkRSXform xform[],
+                         const SkRect tex[],
+                         const SkColor colors[],
+                         int count,
+                         SkBlendMode mode,
+                         const SkSamplingOptions& sampling,
+                         const SkRect* cullRect) = 0;
+  virtual void drawPicture(const sk_sp<SkPicture> picture) = 0;
+  virtual void drawDisplayList(const sk_sp<DisplayList> display_list) = 0;
+  virtual void drawTextBlob(const sk_sp<SkTextBlob> blob,
+                            SkScalar x,
+                            SkScalar y) = 0;
+  // Unfortunately SkDrawShadowRec requires including an internal Skia header
```

Comment:
Add a TODO with a link to the skia bug for this

Reply:
I will if they don't move the struct reference soon.